### PR TITLE
Add cluster template to tenant

### LINKED ISSUE #48: New field for default cluster catalog GitRepo location in Tenant object
Add a new field to the `Tenant` object to allow setting a default location for cluster catalog repositories and specify which secret will be used:

```
spec:
  clusterCatalogGitRepo:
    baseURL: ssh://git@git.example.net/syn/cluster-catalogs/
    apiSecretRef:
      name: git-example-credentials
```

This field can then be used by the API to figure out where the default location of cluster-catalog repositories are stored.

Example: `spec.clusterCatalogGitRepo.baseURL` could be set to `ssh://git@git.example.net/syn/cluster-catalogs/` which is then used by the API to set the corresponding field when creating a cluster without explicitly specifying a location for the cluster catalog git repository.

### PR DESCRIPTION
**Pull request by srueg**
A tenant can specify a cluster template which will be used for newly
created clusters and allows templating.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

Closes #48 and #109 
Supersedes #108

### CODE REVIEW

**Review by corvus-ch — APPROVED** [review]

**Comment by srueg** [comment]
I tested this on the Synfra cluster and it all seems to work so far.
It's deployed in the `lieutenant-dev` namespace, if you want to give it a go.

**Comment by srueg** [comment]
And here are the changes for the API to allow empty GitRepoURL on clusters: https://github.com/projectsyn/lieutenant-api/pull/86

**Review by corvus-ch — CHANGES_REQUESTED** [review]
This needs end user documentation explaining how the templating works and how it can be used.

Further we need a follow up to have templating errors being reflected as conditions and events on the effected objects.

**Comment by srueg** [comment]
Follow up for improved error handling: #112

**Comment by srueg** [comment]
Ok, I wrote an explanation on how to use the templating.

Also I fixed two bugs if the tenant or cluster GitRepoTemplate is empty. Now, no GitRepo object is created (as it was described in the docs).

File: docs/modules/ROOT/pages/explanations/templating.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  displayName: Big Corp.
+  gitRepoTemplate:
+    apiSecretRef:
+      name: vshn-gitlab
+    path: syn-dev/customers
+    repoName: t-aezoo6
+  clusterTemplate:
+    facts:
+      name: '{{ index .Annotations "syn.tools/name"}}'
+      creation: '{{ .CreationTimestamp }}'
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanations/templating.adoc:34]: Please explicitly state that `.metadata.…` fields are embedded and are accessible directly.

File: docs/modules/ROOT/pages/explanations/templating.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    gitRepoTemplate:
+      apiSecretRef:
+        name: secret-{{ .Spec.TenantRef.Name }}
+      path: cluster-catalogs/{{ index .Spec.Facts "cloud"}}
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanations/templating.adoc:38]: Please explicitly state that the `index` function must be used when accessing dicts.

**Review by corvus-ch — CHANGES_REQUESTED** [review]
Examples are good to have but they tend to be insufficient. Too much that is left to interpretation and or might be missed. Please explain the rules of how access fields in more detail.

File: docs/modules/ROOT/pages/explanations/templating.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+== Cluster Template
+
+The tenant field `.spec.clusterTemplate` <<references/crds-html.adoc#syn.tools/v1alpha1.TenantSpec,defines>> a template which is used to set default values for clusters of this tenant.
+All _string_ fields within this allow to use Go templates which are rendered by the operator.
+As input data to the template rendering, the instance of the cluster on which the template is being applied on can be used.
+This data is a Go type defined https://github.com/projectsyn/lieutenant-operator/blob/34db9df5d2090a8840e69816d16ec13527c4fc2c/pkg/apis/syn/v1alpha1/cluster_types.go#L60[here].
+
+Since it's a Go type, only https://golang.org/ref/spec#Exported_identifiers[exported fields] (uppercase) can be accessed.
+
+[INFO]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanations/templating.adoc:16]:
```suggestion
[NOTE]
```

INFO is not a known admonition. See https://asciidoctor.org/docs/user-manual/#admonition.

File: docs/modules/ROOT/pages/explanations/templating.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+====
+Since the `metav1.TypeMeta` (`.metadata` in YAML) is an https://golang.org/doc/effective_go.html#embedding[embedded] struct, you must access it's fields directly (and not via `.Metadata` as in YAML).
+For example the name is accessed with `{{ .Name }}`.
+====
+
+[INFO]
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/explanations/templating.adoc:22]:
```suggestion
[NOTE]
```

**Review by corvus-ch — CHANGES_REQUESTED** [review]

**Review by simu — APPROVED** [review]

**Review by corvus-ch — APPROVED** [review]

**Review by tobru — APPROVED** [review]

**Pull request merged**